app/services/order_item_sanitized_parameters.rb

```diff
@@ -0,0 +1,46 @@
+class OrderItemSanitizedParameters < TopologyServiceApi
+  MASKED_VALUE = "********".freeze
+
+  def process
+    sanitized_parameters
+  rescue StandardError => e
+    Rails.logger.error("OrderItemSanitizedParameters #{e.message}")
+    raise
+  end
+
+  private
+
+  def order_item
+    @order_item ||= OrderItem.find_by!(:id => params[:order_item_id])
+  end
+
+  def service_plan_ref
+    order_item.service_plan_ref
+  end
+
+  def service_parameters
+    order_item.service_parameters
+  end
+
+  def service_plan_schema
+    plan = api_instance.show_service_plan(service_plan_ref)
+    plan.create_json_schema
+  rescue TopologicalInventoryApiClient::ApiError => e
+    Rails.logger.error("DefaultApi->show_service_plan #{e.message}")
+    raise
+  end
+
+  def sanitized_parameters
+    svc_params = ActiveSupport::HashWithIndifferentAccess.new(service_parameters)
+    service_plan_schema[:properties].each_with_object({}) do |(key, attrs), result|
+      value = hide?(key, attrs) ? MASKED_VALUE : svc_params[key]
+      result[attrs[:title]] = value
```

Comment:
We may want to pass all `title`, `key`, and `value` to the approval service. Initially when the content is presented to human descriptive `title` is reasonable. In the future when we need to introduce sophisticated business rules, `key` is more appropriate than `title`.

Reply:
@bzwei 
We can pass the whole parameter json schema but the email recipient or slack message receiver would have to parse the payload before it can display it to the user.
If we display a hash to the user with multiple fields it's going to be confusing to the user. So for now its just using the Title and the value which might be a lot easier to view than the whole hash.